homeassistant/components/generic/config_flow.py

```diff
@@ -314,9 +331,24 @@ async def async_step_user(
                         # The automatically generated still image that stream generates
                         # is always jpeg
                         user_input[CONF_CONTENT_TYPE] = "image/jpeg"
-
-                    return self.async_create_entry(
-                        title=name, data={}, options=user_input
+                    self.cached_user_input = user_input
+                    self.cached_title = name
+
+                    # Register a temporary view so that we can show a preview
+                    flow_id = self.flow_id
+                    preview = GenericCamera(hass, user_input, flow_id, "preview")
+                    register_preview(hass, flow_id, preview)
+                    self.preview_increment += 1
+                    preview_url = f"/api/generic/preview_flow_image/{flow_id}_{self.preview_increment}"
```

Comment:
Yeah i think you are right that changing the url is the best idea for doing this. I was looking at what the home assistant frontend does and it doesn't seem to do anything special here, however, but it is using lit element to update the source explicitly.

- I do think the "random" should be part of the query parameter rather than path, if that works. I don't have a strong reason why, just "feels" more correct to me.
- Using the current timestamp can work so you don't have to keep preview state
- I was reminded that calendar has some random data associated with an access token.  But i'm thinking the config flow id state may be enough that nothing extra is needed. (Just wanted to mention this explicitly in case that isn't right)

Reply:
Good idea.  Done.